sunpy/io/fits.py

```diff
@@ -161,9 +161,11 @@ def write(fname, data, header, hdu_type=None, **kwargs):
         n-dimensional data array.
     header : `dict`
         A header dictionary.
-    hdu_type: `None` or `~astropy.io.fits.CompImageHDU`
-        `None` will return a normal FITS files.
-        `~astropy.io.fits.CompImageHDU` will rice compress the FITS file.
+    hdu_type : `~astropy.io.fits.hdu.base.ExtensionHDU` instance or class, optional
+        By default, a FITS file is written with the map in its primary HDU.
+        If a type is given, a new HDU of this type will be created.
+        If a HDU instance is given, its data and header will be updated from the map.
```

Comment:
This sentence suggests that providing an HDU instance with data is acceptable and will simply result in its data being replaced, rather than raising an error as is currently coded.  Is it problematic to accept an HDU instance with data?  Regardless the documentation needs to match the behavior.

Reply:
It shouldn't be problematic to accept a HDU instance with data. I just thought it would be unlikely that someone would want to do that and if they did, they probably didn't mean to. Thinking this through more, it may be best to just allow for existing data to be present (i.e. implement as currently documented). What do you think?

Reply:
My inclination is to support passing in HDU instances that already have data, although maybe there ought to a be a `deepcopy()` in that case.

Reply:
I've now removed the requirement for no data. Actually, it seems that some data *needs* to be set during initialisation of `CompImageHDU` for the compression settings to be remembered. I opened an issue at https://github.com/astropy/astropy/issues/12216.

Also, I tried wrapping the HDU instance in `copy.deepcopy(...)`, however that causes the tests to fail due to differences in the HDU's header. I think this is an upstream issue and may be related to https://github.com/astropy/astropy/pull/12058 so it might be best if we don't create a copy of the HDU. I think the documentation is clear enough that the HDU instance will simply be updated without copying.